### PR TITLE
@orta=> Return of the Sync Notification

### LINKED ISSUE #154: Show new sync notifications in lab settings
similar to:

looks_correct_with_sync_recommendation 2x

### PR DESCRIPTION
**Pull request by sarahscott**
- [x] Sync badge in the top view controller
- [x] Sync badge in the settings menu
- [x] Sync button turns purple when recommending new syncs
- [x] Lots of tests

Some of the snapshots look very slightly different because I updated a constraint in the settings button nib - it moved the chevrons over by a pixel.

Fixes #154

### CODE REVIEW

**Comment by ArtsyMobile** [comment]
&nbsp; | 1 Error
    ------------- | ------------
        :no_entry_sign: | No CHANGELOG changes made
    

  

  

  


<p align="right" meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/KrauseFx/danger/">danger</a>
</p>

**Comment by orta** [comment]
The changelog thing is covered by https://github.com/KrauseFx/danger/issues/34

File: ArtsyFolio Tests/View Controllers/ARLabSettingsMenuViewModelTests.m [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    });
+});
+
+describe(@"logging out", ^{
+    it(@"tells the app delegate to logout", ^{
+        OCMockObject *mockDelegate = [OCMockObject partialMockForObject:subject.appDelegate];
+    
+        [[mockDelegate expect] startLogout];
+        [subject logout];
+        [mockDelegate verify];
```
Comment (orta) [inline ArtsyFolio Tests/View Controllers/ARLabSettingsMenuViewModelTests.m]: good use of a mock

File: Classes/Controllers/Settings/ARLabSettingsMenuViewModel.h [hunk 2]
```diff
@@ -6,24 +6,27 @@
...
-- (BOOL)shouldEnablePresentationMode;
-
+/// Returns either a settings cog or 'x' depending on device
 - (UIImage *)settingsButtonImage;
 
+/// Logs out via ARAppDelegate
 - (void)logout;
 
+/// Deprecated once old settings are removed
 - (void)switchToOriginalSettings;
```
Comment (orta) [inline Classes/Controllers/Settings/ARLabSettingsMenuViewModel.h]: nice work here too

File: Classes/Controllers/Settings/ARLabSettingsSectionButton.m [hunk 3]
```diff
@@ -47,4 +49,9 @@ - (void)hideChevron
     self.chevron.alpha = 0;
 }
 
+- (void)showAlertBadge:(BOOL)show
+{
+    self.alertBadge.hidden = !show;
+}
```
Comment (orta) [inline Classes/Controllers/Settings/ARLabSettingsSectionButton.m]: this is a nice move too

File: Classes/Controllers/Settings/ARLabSettingsSectionButton.xib [hunk 4]
```diff
@@ -1,9 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" standalone="no"?>
-<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="9059" systemVersion="15B42" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES">
+<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="9531" systemVersion="15C50" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES">
     <dependencies>
         <deployment identifier="iOS"/>
-        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="9049"/>
+        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="9529"/>
         <capability name="Aspect ratio constraints" minToolsVersion="5.1"/>
+        <capability name="Constraints with non-1.0 multipliers" minToolsVersion="5.1"/>
```
Comment (orta) [inline Classes/Controllers/Settings/ARLabSettingsSectionButton.xib]: this is a warning

File: Classes/Controllers/Settings/ARLabSettingsSectionButton.xib [hunk 5]
```diff
@@ -55,63 +55,44 @@
...
-                    <exclude reference="y3T-KH-5hk"/>
-                    <exclude reference="0GB-97-s5d"/>
-                    <exclude reference="bkl-Bp-Gvb"/>
-                    <exclude reference="dH6-Ig-pIH"/>
-                    <exclude reference="q9s-xO-YLg"/>
-                    <exclude reference="x2m-Mc-U6U"/>
-                </mask>
-            </variation>
-            <point key="canvasLocation" x="558" y="446"/>
+            <point key="canvasLocation" x="557.5" y="446"/>
```
Comment (orta) [inline Classes/Controllers/Settings/ARLabSettingsSectionButton.xib]: about this, are you sure you want a `x.5` coordinate?

File: Classes/Controllers/Top View Controller/ARTopViewController.m [hunk 6]
```diff
@@ -124,6 +123,20 @@ - (void)fadeInSwitchView
...
+#pragma mark Sync Notifications Badge
+
+- (void)updateSyncNotificationBadge
+{
+    [self.cmsMonitor checkCMSForUpdates:^(BOOL updated) {
+        if (updated) {
+            [self.toolbarController showSyncNotificationBadge];
+        } else {
+            [self.toolbarController hideSyncNotificationBadge];
+        }
```
Comment (orta) [inline Classes/Controllers/Top View Controller/ARTopViewController.m]: this block will probably end up as in a retain loop, would recommend doing the selfs above as weak

**Comment by orta** [comment]
Cool - just some minor niggles.

**Comment by sarahscott** [comment]
Fixed the wonky constraints and the block in TopVC

**Comment by orta** [comment]
ace

**Pull request merged**